userprofile/serializers.py

```diff
@@ -163,3 +163,9 @@ class HealthCheckUserProfileSerializer(serializers.ModelSerializer):
     class Meta:
         model = HealthCheckUserProfile
         fields = "__all__"
+
+    def update(self, instance, validated_data):
+        if "data" in validated_data:
+            validated_data["data"].update(instance.data)
```

Comment:
Will this not do this wrong way around? eg. if we have `{"a":"b"}` currently on the model, and we want to update it to `{"a":"c"}`, then this code will return `{"a":"b"}`, not the `{"a":"c"}` that we want.

We should probably adjust the unit test to ensure that it overwrites keys that are specified, but leaves existing keys that are not specified.

Reply:
hehe, I'll fix